include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,350 @@
+    {
+        return translate_single_fn_v(std::forward<irng_t>(irange));
+    }
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\brief A view performs single frame translation of nucleotide into aminoacid alphabet
+ * \tparam irng_t The type of the range being processed.
+ * \param irange The range being processed.
+ * \returns A range containing aminoacid sequence. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to translate nucleotide sequences into aminoacid sequences:
+ * 2 frames - forward/ reverse sequence
+ * 3 frames - all forward frames
+ * 6 frames - all forward and reverse frames
+ *
+ * \par View properties
+ *
+ * |                                  |                                                           |
+ * |----------------------------------|-----------------------------------------------------------|
+ * | concepts required of `irng_t`    | `seqan3::random_access_range_concept<irng_t>`             |
+ * |                                  | `&& seqan3::sized_range_concept<irng_t>`                  |
+ * |                                  | `&& seqan3::nucleotide_concept<value_type_t<irng_t>>`     |
+ * | concepts met by `rrng_type`      | `seqan3::random_access_range_concept<rrng_type>`          |
+ * | `value_type_t<rrng_type>`        | `value_type_t<irng_t>`                                    |
+ *
+ * * `irng_t` is the type of the range modified by this view (input).
+ * * `rrng_type` is the type of the range returned by this view.
+ * * for more details, see \ref view.
+ *
+ * \par Example
+ *
+ * Operating on a range of seqan3::dna5:
+ * ```cpp
+ * dna5_vector vec{"ACGTACGTACGTA"_dna5};
+ *
+ * // single frame translation
+ * auto v1 = vec | view::translate_single;                     // == [T,Y,V,R]
+ *
+ * // function syntax
+ * auto v2 = view::translate_single(vec);                      // == [T,Y,V,R]
+ *
+ * // combinability
+ * auto v3 = vec | view::complement | view::translate_single;  // == [C,M,H,A]
+ * ```
+ */
+seqan3::detail::translate_single_fn const translate_single;
+} // namespace seqan3::view
+
+namespace seqan3::detail
+{
+struct translate_frames_fn
+{
+    /*!\brief Reverse, forward-frame or six-frame translation from nucleotide to aminoacid.
+     * \tparam irng_t The type of the range being processed.
+     * \tparam size_type The type of the input number of frames. Must be an integral type.
+     * \param irange The range being processed.
+     * \param num The number of frames to compute  (2, 3 or 6).
+     */
+    template <typename irng_t, typename size_type>
+        requires forward_range_concept<irng_t>
+                 // && nucleotide_concept<ranges::value_type_t<std::decay_t<irng_t>>>
+                 && std::is_integral_v<size_type>
+    auto operator()(irng_t && irange, size_type const & num) const
+    {
+        std::vector<ranges::any_random_access_view<aa27> > frames;
```

Comment:
I am preparing a PR that will introduce a `any_sized_random_access_view` so we don't loose the size here!